src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -902,4 +920,7 @@ void ExecuteAsyncAction()
         RunTestCleanupMethod(timeoutResult);
         return timeoutResult;
     }
+
+    void IDisposable.Dispose()
+        => _timeoutTokenSource?.Dispose();
```

Comment:
I'm curious why is it needed? Don't we dispose in the finally block already in the same method that creates the CTS?

Reply:
I think the CA warning is because we are storing the CTS in a field. But do we actually need to store it in a field? We should be using it in a very limited scope and I guess it should be possible to just pass it around?

Reply:
You are right, I am refactoring.